### PR TITLE
Remove the message correlation guide from 8.5 and above

### PR DESCRIPTION
**Pull request by conceptualshark**
## Description

Relates to https://github.com/camunda/developer-experience/issues/69

Remove the message correlation guide from versions 8.5 and above.

## When should this change go live?

- [ ] This is a bug fix, security concern, or something that needs **urgent release support**.
- [ ] This is already available but undocumented and should be released within a week.
- [ ] This on a **specific schedule** and the assignee will coordinate a release with the DevEx team. (apply `hold` label or convert to draft PR)
- [ ] This is part of a scheduled alpha or minor. (apply alpha or minor label)
- [x] There is **no urgency** with this change and can be released at any time.

## PR Checklist

- [ ] My changes are for an **already released minor** and are in `/versioned_docs` directory.
- [ ] My changes are for the **next minor** and are in `/docs` directory (aka `/next/`).

- [ ] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned the Engineering DRI or delegate.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @camunda/tech-writers as a reviewer.

### CODE REVIEW

File: docs/components/concepts/messages.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ title: "Messages"
 description: "Learn how process instances can respond to incoming messages."
 ---
 
-Process instances can respond to incoming messages. Published messages must be mapped onto a process instance. This step is called **[message correlation](/guides/message-correlation.md)**.
+Process instances can respond to incoming messages. Published messages must be mapped onto a process instance. This step is called message correlation.
```
Comment (christinaausley) [inline docs/components/concepts/messages.md:7]: There are about 90 references of message correlation per version of docs -- perhaps we should create a backlog issue to add this term to the docs glossary and link to it if we are removing the doc on message correlation?

Comment (akeller) [reply docs/components/concepts/messages.md:7]: Good idea, I am ok either doing that in the scope of this PR or a separate one.

Comment (conceptualshark) [reply docs/components/concepts/messages.md:7]: Would it make sense to add to the glossary, or link to message correlation [here](https://docs.camunda.io/docs/next/components/modeler/bpmn/message-events/#message-correlation)?

Comment (christinaausley) [reply docs/components/concepts/messages.md:7]: I think linking to the existing documentation would be fine 👍

**Comment by conceptualshark** [comment]
@christinaausley Updated this to replace the old link to the message correlation section in Message events - I think it should be good to go now.

**Review by christinaausley — APPROVED** [review]

**Review by akeller — APPROVED** [review]
Thank you. I know we don't enjoy removing guides, but it's the right thing to do in this case. Thanks for adjusting!

**Pull request merged**